### ISSUE TITLE
AP is -1.000

### ISSUE THREAD

**Issue by wgqbit95** [zylo117/Yet-Another-EfficientDet-Pytorch#401]
![2](https://user-images.githubusercontent.com/34861661/85085024-1c737580-b209-11ea-9a36-1ccee59a5ad6.png)

Could you please help me to find out the problem.

**Comment by wgqbit95** [comment]
![3](https://user-images.githubusercontent.com/34861661/85085235-cb17b600-b209-11ea-97b8-e404f403202f.png)
I think the training process is normal, isn't it？plz，help me（o(╥﹏╥)o）

**Comment by zylo117** [comment]
keep training

**Comment by wgqbit95** [comment]
After 140 epoch， mAP on val dataset is still -1. 
mAP on train set is around 0.5
Is it caused by the k-means result.(Since I haven't change the default para about anchor ratios in the code)

**Comment by zylo117** [comment]
refer to the tutorial

**Comment by wgqbit95** [comment]
lol alright

**Comment by petinhoss7** [comment]
> 
> 
> After 140 epoch， mAP on val dataset is still -1.
> mAP on train set is around 0.5
> Is it caused by the k-means result.(Since I haven't change the default para about anchor ratios in the code)

Where should I make the changes? I get as well -1 everytime :/

**Comment by wgqbit95** [comment]
> > After 140 epoch， mAP on val dataset is still -1.
> > mAP on train set is around 0.5
> > Is it caused by the k-means result.(Since I haven't change the default para about anchor ratios in the code)
> 
> Where should I make the changes? I get as well -1 everytime :/

I am still looking for where the problem is, still not solved. o(╥﹏╥)o

**Comment by zylo117** [comment]
probably dataset category id doesn't match the requirement

**Comment by wgqbit95** [comment]
> > After 140 epoch， mAP on val dataset is still -1.
> > mAP on train set is around 0.5
> > Is it caused by the k-means result.(Since I haven't change the default para about anchor ratios in the code)
> 
> Where should I make the changes? I get as well -1 everytime :/

First, you can try this repo to regenerate the coco format json file. Then, check the optimise method. I met the problem when i use SGD optimization method. I changed SGD to AdamW, now.
And now, the AP is somewhat normal.

**Comment by wgqbit95** [comment]
> > > After 140 epoch， mAP on val dataset is still -1.
> > > mAP on train set is around 0.5
> > > Is it caused by the k-means result.(Since I haven't change the default para about anchor ratios in the code)
> > 
> > 
> > Where should I make the changes? I get as well -1 everytime :/
> 
> I am still looking for where the problem is, still not solved. o(╥﹏╥)o
https://github.com/ChangeZH/ALL2COCO This repo to regenerate the json file of your dataset.

**Comment by zylo117** [comment]
> > > After 140 epoch， mAP on val dataset is still -1.
> > > mAP on train set is around 0.5
> > > Is it caused by the k-means result.(Since I haven't change the default para about anchor ratios in the code)
> > 
> > 
> > Where should I make the changes? I get as well -1 everytime :/
> 
> First, you can try this repo to regenerate the coco format json file. Then, check the optimise method. I met the problem when i use SGD optimization method. I changed SGD to AdamW, now.
> And now, the AP is somewhat normal.

So, in that case, I would say that it is underfiting when training with SGD, it's not yet converged, not even close, so the results will all be invalid.

**Comment by liufengwei520** [comment]
the question where I met as well  when training with Adamw . My training loss is 0.3 about at last ,is it not enough to train?

**Comment by zylo117** [comment]
@liufengwei520 not yet

**Comment by liufengwei520** [comment]
还是不行，我测试各个图片框，发现都框中了，但是有许多没有框中的

**Comment by mytk2012** [comment]
> 还是不行，我测试各个图片框，发现都框中了，但是有许多没有框中的

How should I do to viusalize the results of test？

**Comment by qubapikaqi** [comment]
我姓请问一下这个map是训练完自动打印还是要自己手动改代码呀，因为我训练的时候没有打印出来的map，只有loss

**Comment by qubapikaqi** [comment]
请问map是训练完打印出来的嘛，还是要自己改一下代码呀，因为我训练的时候没有打印出来的map，而且我换成voc数据集训练之后，保存下来的checkpoint经常会是空文件，我想问一下我可能哪里出错了呢

**Comment by zylo117** [comment]
@qubapikaqi run `coco_eval.py` following the instructions to get mAP results. It's written on readme. And you should run through tutorial first.
